File: app/src/main/java/com/coinninja/coinkeeper/model/db/enums/AccountStatus.java
package com.coinninja.coinkeeper.model.db.enums;

import org.jetbrains.annotations.Nullable;

public enum AccountStatus {
    UNVERIFIED(0),
    PENDING_VERIFICATION(10),
    VERIFIED(100);

    public final int id;

    AccountStatus(int id) {
        this.id = id;
    }

    public static AccountStatus from(@Nullable String status) {
        if (status == null) return UNVERIFIED;

        switch (status.toLowerCase()) {
            case "verified":
                return VERIFIED;
            case "pending-verification":
                return PENDING_VERIFICATION;
        }
        return UNVERIFIED;
    }

    public static String asString(AccountStatus status) {
        switch (status) {
            case VERIFIED:
                return "verified";
            case PENDING_VERIFICATION:
                return "pending-verification";
            default:
                return "unverified";
        }
    }

    public int getId() {
        return id;
    }
}

File: app/src/main/java/com/coinninja/coinkeeper/model/db/enums/AccountStatus.kt
package com.coinninja.coinkeeper.model.db.enums

enum class AccountStatus constructor(val id: Int) {
    UNVERIFIED(0),
    PENDING_VERIFICATION(10),
    VERIFIED(100);

    companion object {

        fun from(status: String?): AccountStatus {
            if (status == null) return UNVERIFIED

            when (status.toLowerCase()) {
                "verified" -> return VERIFIED
                "pending-verification" -> return PENDING_VERIFICATION
            }
            return UNVERIFIED
        }

        fun asString(status: AccountStatus): String {
            when (status) {
                VERIFIED -> return "verified"
                PENDING_VERIFICATION -> return "pending-verification"
                else -> return "unverified"
            }
        }
    }
}
